### ISSUE TITLE
Properly installing some dependency packages

### ISSUE THREAD

**Issue by JILPulvino** [2i2c-org/pilot#66]
We'd like to install a few packages for all our users:

- [ ] nodejs with the ability to manage node versions
- [ ] gatsby
- [x] A C and a C++ compiler as mentioned in the requirements for this R library (https://igraph.org/r/)
- [x] igraph
- [x] devtools for R
- [x] Rcpp
- [x] roxygen2
- [x] rJava
- [x] davidsjoberg/ggsankey

There are other R and python packages we'd like to system wide install, but happy to do that myself if you can say how. The above are ones that have just caused me headaches in the past installing on linux systems.

**Comment by JILPulvino** [comment]
additional request - is it possible to install alternative versions of R? It seems part of the issue with igraph is that a package has not been updated to version 4.x and is still reliant on version 3.6

**Comment by yuvipanda** [comment]
Hey @JILPulvino. Do you have experiences with building docker images yourselves? I think giving you control over that is the easiest thing to do long term - https://pilot.2i2c.org/en/latest/admin/howto/environment.html#bring-your-own-docker-image has some docs, but I can work closer in setting it up. This gives you full control over the environment, and you can update it without blocking on 2i2c staff in any way. It shouldn't require understanding Dockerfiles or similar if you don't want to.

If you don't want to do that yet, I can add these packages to the current base image. But medium / long term, we'll make a new image for you.

**Comment by JILPulvino** [comment]
Thank you! I don't have experience building my own. I don't mind the idea of maintaining one, but am a little intimidated about setting up initially. I'll read the docs and maybe experiment some on a toy repo and get back to you with challenges.

**Comment by JILPulvino** [comment]
@yuvipanda I've gone through much of the 2i2c documentation and better understand how I can bypass learning everything about docker. With that in mind, I created a binder folder with an environment.yml and an install.R file for now. Before testing those out and using the repo2docker tool, I'm wondering if you can point me in the direction of the type of configuration file I would need to install [gatsby](https://www.gatsbyjs.com/docs/tutorial/part-zero/) and node/npm?

**Comment by yuvipanda** [comment]
So images built with repo2docker already contain node + npm. Based on my understanding from https://github.com/jupyterhub/repo2docker/pull/573, you should be able to do `npm install -g gatsby-cli` in your `postBuild` file should do the trick.

**Comment by JILPulvino** [comment]
Nevermind, I think I found the issue - a few of the packages I was trying to install in the environment.yml file are not in conda forge

**Comment by yuvipanda** [comment]
Can you give me a link to your repo?

**Comment by JILPulvino** [comment]
@yuvipanda I tried the `jupyter-repo2docker` on the original shared repo and ran into the same error as before. So I created a new repo (added you to it) and just copied over the environment.yml and install.r example files from the pilot-hubs repo to try and run `jupyter-repo2docker`. I ran into the same error.

**Comment by yuvipanda** [comment]
ok, let's unblock you by just installing these packages into our base image for now - and then we can try figure out a separate environment story. How does that sound, @JILPulvino?

@GeorgianaElena can you add the packages requested here to the base pilot hubs image?

**Comment by JILPulvino** [comment]
Yep, that works for me. The only additional one is 
`devtools::install_github("davidsjoberg/ggsankey")`

You mentioned that ultimately it might be best for me to just maintain a docker image. Is that still your recommendation? I've been muddling through a docker tutorial, but just wondering if it makes sense to keep going that route or to work on trying to get the docker repo I just created up and running via `jupyter-repo2docker` instead?

**Comment by GeorgianaElena** [comment]
Hey @JILPulvino :wave:! I updated your top comment and checked the boxes for the pkgs that were installed into the base image. I hope that's ok. Please let me know if you spot issues with any of it.

**Comment by JILPulvino** [comment]
Thank you!

**Comment by choldgraf** [comment]
Hey all - we've got two unchecked boxes at the top. Do we still need to take action here or are we good to go?

**Comment by choldgraf** [comment]
sounds good - that seems like a useful skillset to get comfortable with anyway, especially since repo2docker makes it relatively straightforward to do so!

**Comment by choldgraf** [comment]
@JILPulvino I'm going to close this issue since I don't think there's anything actionable on 2i2c's end. If you run into major issues with using your own Docker image, we can discuss in another issue or thread. That work? (feel free to re-open if you disagree and would like to keep tracking something here)

**Comment by JILPulvino** [comment]
@choldgraf the only reason I can see for leaving this open moving forward is that I previously wasn't able to get repo2docker to run - thus the pursuit of creating a docker image from scratch. Perhaps I should open an issue on the repo2docker repo instead though?

**Comment by choldgraf** [comment]
Ah that makes sense - if you run into issues maybe you can open an issue about that either in repo2docker or ask around here? It'll be easier to spot if it's explicitly scoped to repo2docker and usch
